### PR TITLE
SNOW-1347401 Fallback cleanup: string accessors

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1347401

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Raise NotImplementedError for all Series.str API that use fallback pandas code.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 1]
```diff
@@ -1039,18 +1073,23 @@ def capitalize(self):
         return Series(query_compiler=self._query_compiler.str_capitalize())
 
     def swapcase(self):
+        ErrorMessage.not_implemented("Series.str.swapcase() is not implemented")
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1076]: one suggestion (no need to follow through): we could also have a @not_implemented decorator, that automatically take the function name and issue the string. Then all messages are formatted the same and you do not need to manually edit all of them :)

**Review by sfc-gh-lspiegelberg — APPROVED** [review]

**Review by sfc-gh-mvashishtha — APPROVED** [review]

File: src/snowflake/snowpark/modin/pandas/series_utils.py (hunk 1 above)
Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/modin/pandas/series_utils.py:1076]: That's a great idea! I tried it with the decorator raising the error but I don't think it executes as expected with tests. I think I will stick with what I have for now though, thanks!

**Comment by sfc-gh-mvashishtha** [comment]
@sfc-gh-vbudati let's wait to merge this PR till we have the decorators we agreed [here](https://snowflake.slack.com/archives/C04HF38JFAQ/p1714508688696499) to add in SNOW-1341619

File: tests/integ/modin/series/test_str_accessor.py [hunk 2]
```diff
@@ -221,9 +221,10 @@ def test_str_replace_neg(pat, n, repl, error):
         snow_ser.str.replace(pat=pat, repl=repl, n=n)
 
 
+@pytest.mark.xfail(reason="Series.str.split() is not implemented", strict=True)
```
Comment (sfc-gh-joshi) [inline tests/integ/modin/series/test_str_accessor.py:224]: I don't think this should be an XFAIL, we should have `pytest.raises(NotImplementedError)` instead.

File: tests/unit/modin/test_series_strings.py
Comment (sfc-gh-joshi) [inline tests/unit/modin/test_series_strings.py:1]: We shouldn't be removing this file wholesale; we should keep the tests around and check that methods correctly raise `NotImplementedError` instead.

**Review by sfc-gh-joshi — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-mvashishtha** [comment]
> let's wait to merge this PR till we have the decorators we agreed [here](https://snowflake.slack.com/archives/C04HF38JFAQ/p1714508688696499) to add in [SNOW-1341619](https://snowflakecomputing.atlassian.net/browse/SNOW-1341619)

[never mind](https://snowflake.slack.com/archives/C04HF38JFAQ/p1714520035693639?thread_ts=1714508688.696499&cid=C04HF38JFAQ), we don't have to wait for the decorators.

File: tests/unit/modin/test_series_strings.py
Comment (sfc-gh-vbudati) [reply tests/unit/modin/test_series_strings.py:1]: added file back

File: tests/integ/modin/series/test_str_accessor.py [hunk 3]
```diff
@@ -276,6 +290,15 @@ def test_str_no_params(func):
     )
 
 
+@sql_count_checker(query_count=0)
+def test_str_isdigit_negative():
```
Comment (sfc-gh-joshi) [inline tests/integ/modin/series/test_str_accessor.py:294]: Is this a regression? The above test with the "isdigit" parameter isn't XFAILed and has no fallbaack query count.

**Review by sfc-gh-joshi — APPROVED** [review]
Mostly LGTM, just one question.

File: tests/integ/modin/series/test_str_accessor.py (hunk 3 above)
Comment (sfc-gh-vbudati) [reply tests/integ/modin/series/test_str_accessor.py:294]: Ooh good catch, I think I missed the `snowflake_query_compiler` implementation. Reverting this.

**Pull request merged**